### ISSUE TITLE
Why loopy in the example?

### ISSUE THREAD

**Issue by linhaojia13** [mbforbes/py-factorgraph#7]
In my understanding, the loopy BP is suitd to cyclic graph while the BP is suited to acyclic graph. 
The loop BP will conduct the message passing in the cyclic graph repeatedly until convergence, while the BP only conduct one forward propagation and one backpropagation between all leafs and one randomly selected root.
In your example, the factor graph contains two RVs and two factors, which present as an acyclic graph, but it seems that you conduct a loopy BP algorithm as follow:
```
        while cur_iter < max_iters and not converged and not E_STOP:
            # Bookkeeping
            cur_iter += 1

            if progress:
                # self.print_messages(nodes)
                logger.debug('\titeration %d / %d (max)', cur_iter, max_iters)

            # Comptue outgoing messages:
            converged = True
            for n in nodes:
                n_converged = n.recompute_outgoing(normalize=normalize)
                converged = converged and n_converged
```

 Is there anything wrong with my understanding on the belief propagation algorithm or the example?

**Comment by mbforbes** [comment]
Your understanding is correct. The reason I implemented it this way is:

1. I knew the graphs for my project would have cycles, so I only implemented LBP
2. I learned that there's no harm in running LBP w/ multiple iterations if there aren't cycles, because it will simply converge

Thus, for simplicity, it just runs LBP on everything, and acyclic graphs should simply converge immediately.

Does this make sense?
